src/main/java/nl/altindag/gatekeeper/Gatekeeper.java

```diff
@@ -34,29 +34,37 @@ public static void ensureCallerIsAnyOf(Class... allowedCallerClasses) {
             throw new IllegalArgumentException("At least one allowed caller class should be present");
         }
 
-        StackTraceElement[] stackTrace = new Exception().getStackTrace();
-        StackTraceElement caller = stackTrace[2];
-        StackTraceElement target = stackTrace[1];
-
-        boolean isCallerAllowedToCallTarget = isCallerAllowedToCallTarget(allowedCallerClasses, caller.getClassName());
+        Class<?> caller = internalSecurityManager.getCallerClassName(3);
+        boolean isCallerAllowedToCallTarget = isCallerAllowedToCallTarget(allowedCallerClasses, caller);
 
         if (!isCallerAllowedToCallTarget) {
+            StackTraceElement[] stackTrace = new Exception().getStackTrace();
```

Comment:
It would be nice if we can inline this one. We don't need the stacktrace, but just only the target. What do you think?